app/api/v1/user/service.js

```diff
@@ -0,0 +1,54 @@
+const jwt = require('../../../utils/jwt');
+
+// Expose User model
+const User = require('../../../models/user');
+
+const passwordStars = 'xxxxxxxxxxxx';
+
+module.exports = {
+  validatePassword(user, password, res) {
+    user.isValidPassword(password).then((isValid) => {
+      if (isValid) {
+        // generate api token that will be expired in 24 hours
+        const token = jwt.generate({ username: user.name });
+        res.json({
+          success: true,
+          _id: user._id,
+          token,
+        });
+      } else {
+        res.json({ success: false, message: 'Wrong password' });
+      }
+    });
+  },
+  verifyToken(res, token, next) {
+    jwt.verify(token, (err, userPayload) => {
+      if (err || userPayload === undefined) {
+        res.status(401).json({ success: false, message: 'invalid token' });
+      } else {
+        // valid token
+        User.findOne({
+          name: userPayload.username,
+        }, (userError, user) => {
+          if (userError) throw userError;
```

Comment:
I'm not sure the Error thrown here will be caught somewhere else.

Reply:
Sorry, my mistake. I will fix it.

Reply:
Fixed